src/cy-reporter.js

```diff
@@ -36,6 +36,39 @@ var resultsFolder;
  * @returns {testcase record object}
  */
 
+function setConfiguration(options) {
+
+  console.debug('START: configuration & options:');
+
+  resultsFolder = 'results';
+  logsFolders = path.join('cypress', 'logs');
+
+  // Add error checking!!
+  const JSON_FILE = "__config.json";
+  try {
+    const jsonConfig = fs.readFileSync(JSON_FILE);
+    const objConfig  = JSON.parse(jsonConfig);
+    var specPattern = objConfig.resolved.specPattern.value;
+    var indx = specPattern.indexOf("/**/");
```

Comment:
Does cypress guarantee the `specPattern` to be always set (e.g. not null)?
Is it guaranteed to always contain `/**/`? I think, you should be looking for just `/*`.

Reply:
Using the `specPattern` has challenges, as discussed. Also, the exact location of videos, and screenshots can vary based on the run options. For example - `cypress run --spec cypress/e2e/folder/the.spec.cy.js` will generate a video with **no** *folder* in the path.  Going to experiment with using the plugin.